### PR TITLE
[FRONT-211] Refactor DU stats & secrets loading

### PR DESCRIPTION
**Pull request by juhah**
Refactor loading of stats & secrets so that they are not always reloaded when switching between edit and preview mode. See comments for details.

### CODE REVIEW

File: app/src/marketplace/containers/EditProductPage/Preview.jsx [hunk 1]
```diff
@@ -141,9 +161,9 @@ const DataUnionStats = () => {
                 id: 'subscribers',
                 value: '0',
             }, {
-                id: 'adminFee',
+                id: 'revenueShare',
                 unit: '%',
-                value: adminFee ? (adminFee * 100).toFixed(0) : '0',
+                value: adminFee ? ((1 - adminFee) * 100).toFixed(0) : '0',
```
Comment (juhah) [inline app/src/marketplace/containers/EditProductPage/Preview.jsx:166]: This is the default value when DU is not yet deployed.

File: app/src/marketplace/containers/EditProductPage/index.jsx [hunk 2]
```diff
@@ -107,19 +109,41 @@ const EditProductPage = ({ product }: { product: Product }) => {
...
+    }, [
+        isDataUnion,
+        loadEthIdentities,
+    ])
+
+    useEffect(() => {
+        if (isDeployed) {
+            loadDataUnion(beneficiaryAddress)
+            loadDataUnionStats(beneficiaryAddress)
+            loadDataUnionSecrets(beneficiaryAddress)
```
Comment (juhah) [inline app/src/marketplace/containers/EditProductPage/index.jsx:132]: TODO: Use same stats loading from `useDataUnionServerStats()` hook and remove this controller method.

File: app/src/marketplace/containers/EditProductPage/Preview.jsx [hunk 3]
```diff
@@ -106,16 +107,35 @@ const Description = () => {
...
-    const { created, adminFee } = product
+    const { subscriberCount } = contractProduct || {}
+    const { created, adminFee, dataUnionDeployed, beneficiaryAddress } = product
 
-    const isDataUnion = !!(product && isDataUnionProduct(product))
-    const isDuDeployed = !!isDataUnion && !!product.dataUnionDeployed && isEthereumAddress(product.beneficiaryAddress)
+    const isDuDeployed = !!dataUnionDeployed && isEthereumAddress(beneficiaryAddress)
 
-    const { stats, memberCount } = useDataUnionStats()
+    const { startPolling, stopPolling, totalEarnings, memberCount } = useDataUnionServerStats()
```
Comment (juhah) [inline app/src/marketplace/containers/EditProductPage/Preview.jsx:117]: Polling and getting the stats array is now two different hooks.

File: app/src/marketplace/containers/ProductPage/useDataUnionServerStats.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        if (!dataUnionId) {
+            throw new Error('Data union id not given')
+        }
+
+        try {
+            const result = await getDataUnionStats(dataUnionId)
+            if (!isMounted()) {
+                return
+            }
+            setTotalEarnings(result.totalEarnings)
```
Comment (juhah) [inline app/src/marketplace/containers/ProductPage/useDataUnionServerStats.js:28]: TODO: Add these values to redux store

File: app/src/marketplace/containers/Products/index.jsx [hunk 5]
```diff
@@ -47,7 +47,7 @@ const Products = () => {
     const { api: createProductModal } = useModal('marketplace.createProduct')
 
     const loadCategories = useCallback(() => dispatch(getCategories(false)), [dispatch])
-    const { load: loadDataUnions, members } = useMemberStats()
+    const { load: loadDataUnions, members } = useAllDataUnionStats()
```
Comment (juhah) [inline app/src/marketplace/containers/Products/index.jsx:50]: The `useAllDataUnionStats` hook can be remove after DU 2.0 functionality is merged

File: app/src/marketplace/modules/dataUnion/hooks/useDataUnionSecrets.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+import { selectDataUnionSecrets } from '../selectors'
+import { getSecrets, postSecret, putSecret, deleteSecret } from '../services'
+import useEntities from '$shared/hooks/useEntities'
+
+function useDataUnionSecrets() {
+    const dispatch = useDispatch()
+    const secrets = useSelector(selectDataUnionSecrets)
+    const { update } = useEntities()
+
+    const load = useCallback(async (dataUnionId) => {
```
Comment (juhah) [inline app/src/marketplace/modules/dataUnion/hooks/useDataUnionSecrets.js:16]: These methods basically act as redux thunk methods

File: app/src/marketplace/modules/dataUnion/services.js [hunk 7]
```diff
@@ -233,26 +233,24 @@ export const getSecrets = ({ dataUnionId }: GetSecrets): ApiResult<Array<Secret>
 type PostSecrect = {
     dataUnionId: DataUnionId,
     name: string,
-    secret: string
 }
 
-export const postSecret = ({ dataUnionId, name, secret }: PostSecrect): ApiResult<Secret> => post({
+export const postSecret = ({ dataUnionId, name }: PostSecrect): ApiResult<Secret> => post({
```
Comment (juhah) [inline app/src/marketplace/modules/dataUnion/services.js:238]: Looks like `secret` is not used

File: app/src/marketplace/modules/dataUnion/hooks/useDataUnionSecrets.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        try {
+            const response = await getSecrets({
+                dataUnionId,
```
Comment (mondoreale) [inline app/src/marketplace/modules/dataUnion/hooks/useDataUnionSecrets.js:19]: When you deploy a fresh DU, `dataUnionId` is `null`.

**Review by mondoreale — APPROVED** [review]
Looks good to me. Not much to add. 👍

File: app/src/marketplace/modules/dataUnion/hooks/useDataUnionSecrets.js (hunk 8 above)
Comment (juhah) [reply app/src/marketplace/modules/dataUnion/hooks/useDataUnionSecrets.js:19]: Fixed in https://github.com/streamr-dev/streamr-platform/pull/1148/commits/ed2ffdf103711498773805fde647abbb31b3d61b

The effect reacted to the changing `beneficiaryAddress`, we should only load it when entering the page. Also added a check for join part stream creation fail.

**Review by mondoreale — APPROVED** [review]

**Review by tumppi — APPROVED** [review]
Good stuff! Had a go with this and seems to work really well 👍

**Pull request merged**